pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -97,20 +96,14 @@ func fixBundleExtracted(bundlePath string, preset crcpreset.Preset) func() error
 				return fmt.Errorf("%s is invalid or missing, run 'crc setup' to download the bundle", bundlePath)
 			}
 		}
+
+		isDefaultBundle := false
 		if bundlePath == constants.GetDefaultBundlePath(preset) {
-			logging.Infof("Downloading %s", constants.GetDefaultBundle(preset))
-			// In case of OKD or podman bundle then pull the bundle image from quay
-			// otherwise use mirror location to download the bundle.
-			if preset == crcpreset.OKD || preset == crcpreset.Podman {
-				if err := image.PullBundle(preset, ""); err != nil {
-					return err
-				}
-			} else {
-				if err := bundle.DownloadDefault(preset); err != nil {
-					return err
-				}
-			}
-			bundlePath = constants.GetDefaultBundlePath(preset)
+			isDefaultBundle = true
+		}
+		var err error
+		if bundlePath, err = bundle.Download(preset, bundlePath, isDefaultBundle); err != nil {
+			return err
```

Comment:
`Remove isDefault argument from bundle.Download`  and ` Extend -b option to accept docker/http/https scheme ` can be squashed together imo.